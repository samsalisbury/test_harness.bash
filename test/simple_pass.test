#!/usr/bin/env bash

# These are the fundamental tests. This test file itself is tested specially, ensuring
# that all tests are run and that we exit with a zero exit code, both when running this
# by itself, and when running with ./testing.bash.

set -euo pipefail

# shellcheck disable=SC1090
source "${TESTING_BASH:-../testing.bash}"

TestOK_Empty() {
  :
}

TestOK_True() {
  true
}

TestOK_Run_True() {
  run true
}

TestOK_Run_Echo() {
  run echo 1
}

TestOK_skip_empty() {
  :
  skip "empty"
}

TestOK_skip_true() {
  true
  skip "true"
}

TestOK_skip_run_true() {
  run true
  skip "run true"
}

TestOK_skip_run_echo() {
  run echo 1
  skip "run echo"
}

TestOK_log() {
  log "Hello world."
}

TestOK_loghelper() {
  log "You will see this."
  loghelper
  nested_loghelper
  log "You will see this."
}

TestOK_skiphelper() {
  log "You will see this."
  skiphelper
  log "You won't see this."
}

nested_loghelper() {
  helper
  log "Logs from nested helpers appear to come from the same line."
  log "See, this line and the next from a nested helper are all the same."
  loghelper
}

loghelper() {
  helper
  log "This line number is where loghelper was called."
}

skiphelper() {
  helper
  skip "This line number is where skiphelper was called."
}
