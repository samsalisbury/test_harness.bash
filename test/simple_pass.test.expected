=== RUN   simple_pass/TestOK_Empty
--- PASS: simple_pass/TestOK_Empty
--- PASS: simple_pass/TestOK_Empty
=== RUN   simple_pass/TestOK_True
--- PASS: simple_pass/TestOK_True
--- PASS: simple_pass/TestOK_True
=== RUN   simple_pass/TestOK_Run_True
--- PASS: simple_pass/TestOK_Run_True
    $ true
--- PASS: simple_pass/TestOK_Run_True
    $ true
=== RUN   simple_pass/TestOK_Run_Echo
--- PASS: simple_pass/TestOK_Run_Echo
    $ echo 1
    1
--- PASS: simple_pass/TestOK_Run_Echo
    $ echo 1
    1
=== RUN   simple_pass/TestOK_skip_empty
--- SKIP: simple_pass/TestOK_skip_empty
    simple_pass.test:30: empty
--- SKIP: simple_pass/TestOK_skip_empty
    simple_pass.test:30: empty
=== RUN   simple_pass/TestOK_skip_true
--- SKIP: simple_pass/TestOK_skip_true
    simple_pass.test:35: true
--- SKIP: simple_pass/TestOK_skip_true
    simple_pass.test:35: true
=== RUN   simple_pass/TestOK_skip_run_true
--- SKIP: simple_pass/TestOK_skip_run_true
    $ true
    simple_pass.test:40: run true
--- SKIP: simple_pass/TestOK_skip_run_true
    $ true
    simple_pass.test:40: run true
=== RUN   simple_pass/TestOK_skip_run_echo
--- SKIP: simple_pass/TestOK_skip_run_echo
    $ echo 1
    1
    simple_pass.test:45: run echo
--- SKIP: simple_pass/TestOK_skip_run_echo
    $ echo 1
    1
    simple_pass.test:45: run echo
=== RUN   simple_pass/TestOK_log
--- PASS: simple_pass/TestOK_log
    simple_pass.test:49: Hello world.
--- PASS: simple_pass/TestOK_log
    simple_pass.test:49: Hello world.
=== RUN   simple_pass/TestOK_loghelper
--- PASS: simple_pass/TestOK_loghelper
    simple_pass.test:53: You will see this.
    simple_pass.test:54: This line number is where loghelper was called.
    simple_pass.test:55: Logs from nested helpers appear to come from the same line.
    simple_pass.test:55: See, this line and the next from a nested helper are all the same.
    simple_pass.test:55: This line number is where loghelper was called.
    simple_pass.test:56: You will see this.
--- PASS: simple_pass/TestOK_loghelper
    simple_pass.test:53: You will see this.
    simple_pass.test:54: This line number is where loghelper was called.
    simple_pass.test:55: Logs from nested helpers appear to come from the same line.
    simple_pass.test:55: See, this line and the next from a nested helper are all the same.
    simple_pass.test:55: This line number is where loghelper was called.
    simple_pass.test:56: You will see this.
=== RUN   simple_pass/TestOK_skiphelper
--- SKIP: simple_pass/TestOK_skiphelper
    simple_pass.test:60: You will see this.
    simple_pass.test:61: This line number is where skiphelper was called.
--- SKIP: simple_pass/TestOK_skiphelper
    simple_pass.test:60: You will see this.
    simple_pass.test:61: This line number is where skiphelper was called.
PASS
ok        simple_pass
